wsi/drm/drmdisplay.cpp

```diff
@@ -419,30 +408,81 @@ bool DrmDisplay::SetBroadcastRGB(const char *range_property) {
   return true;
 }
 
-void DrmDisplay::SetHDCPState(HWCContentProtection state,
+bool DrmDisplay::CheckHDCPStatus() {
+  int val = -1;
+  uint32_t i;
+  ScopedDrmObjectPropertyPtr cprops(drmModeObjectGetProperties(
+      gpu_fd_, connector_, DRM_MODE_OBJECT_CONNECTOR));
+  for (i = 0; i < cprops->count_props; i++) {
+    if (cprops->props[i] == hdcp_id_prop_) {
+      val = cprops->prop_values[i];
+      break;
+    }
+  }
+
+  if ((current_protection_support_ && val == HWCContentProtection::kEnabled) ||
+      (!current_protection_support_ && val == HWCContentProtection::kDesired))
```

Comment:
when and how the status will change from kDesired to kEnabled? Is it possible in the sleep time frame it is already changed to kEnabled, you can not get the kDesired state.

Reply:
sorry, could you please explain more about the sleep time frame?

Reply:
I mean in the SetHDCPState() there is  'usleep(100 * 1000);' for every iteration. is it possible in this time frame, the state changes from kUnDisired->kDesired->kEnabled? for such case, your logic in Check state will not return true.